### PR TITLE
Feat stills

### PR DESCRIPTION
**Pull request by mixcas**

### CODE REVIEW

File: css/site.styl [hunk 1]
```diff
@@ -649,8 +651,10 @@ iframe
   .u-fixed-ratio-content
     transition: background-color $basic-animation-speed
   &:hover .u-fixed-ratio-content
-    background-color: rgba(0,0,0,0.5)
+    background-color: rgba(0,0,0,0.5)d
```
Comment (varffs) [inline css/site.styl]: this looks like a typo? the `d`

File: lib/meta-boxes.php [hunk 2]
```diff
@@ -1,12 +1,11 @@
 <?php
-
 /* Get post objects for select field options */
 function get_post_objects( $query_args ) {
   $args = wp_parse_args( $query_args, array(
-      'posts_per_page' => -1,
-    ) );
+    'post_type' => 'post',
```
Comment (varffs) [inline lib/meta-boxes.php]: why no more posts_per_page here?

File: lib/meta-boxes.php [hunk 3]
```diff
@@ -1,12 +1,11 @@
...
+  ) );
   $posts = get_posts( $args );
-  $post_options = array(" " => null);
+  $post_options = array();
```
Comment (varffs) [inline lib/meta-boxes.php]: why no blank first option here?

File: single-director.php [hunk 4]
```diff
@@ -175,6 +176,26 @@
...
     </section>
+<?php
+$stills = get_post_meta( get_the_ID(), '_igv_stills', true );
+//pr($stills);
+if (!empty($stills)) {
+?>
+    <section id="director-stills" class="director-section u-cf">
+<?php
+  foreach( $stills as $still ) {
+    $still_image = wp_get_attachment_image($still['still_image_id'], 'stills');
```
Comment (varffs) [inline single-director.php]: better to get the html so we get the auto srcset

File: css/site.styl (hunk 1 above)
Comment (mixcas) [inline css/site.styl]: yeah u_u

File: lib/meta-boxes.php (hunk 2 above)
Comment (mixcas) [inline lib/meta-boxes.php]: idk, it was like this

File: lib/meta-boxes.php [hunk 5]
```diff
@@ -43,180 +45,207 @@ function cmb_sample_metaboxes( array $meta_boxes ) {
...
+  $about_page = get_page_by_title('About');
+
+  $about_metabox = new_cmb2_box( array(
+    'id'           => $prefix . 'metabox',
+    'title'        => __( 'About Page Metabox', 'cmb2' ),
+    'object_types' => array( 'page', ),
+    'context'      => 'normal',
+    'priority'     => 'high',
+    'show_names'   => true,
+    'show_on'      => array( 'id' => array( $about_metabox->ID ) ),
```
Comment (varffs) [inline lib/meta-boxes.php]: this should be `$about_page->ID`

File: single-director.php (hunk 4 above)
Comment (mixcas) [inline single-director.php]: exactly… this is doing that

**Comment by varffs** [comment]
Seems good other than the comments. 2 things:

- The Stills menu item shows even if there are no stills
- The UX of the repeating field file upload is pretty rubbish. Is it really not possible to parse the shortcode and use a wysiwyg with WP Gallery?

File: lib/meta-boxes.php (hunk 2 above)
Comment (mixcas) [inline lib/meta-boxes.php]: oh no, i updated this function from the boilerplate…
maybe i shouldnt
i'll check

Comment (mixcas) [inline lib/meta-boxes.php]: yeah, the -1 is missing, without it doesn't work as it should
i think `'posts_per_page' => -1` should be in the boilerplate

File: lib/meta-boxes.php (hunk 3 above)
Comment (mixcas) [inline lib/meta-boxes.php]: this was updated from the boilerplate too
that array (old one) seems just wrong tho
:)

**Comment by mixcas** [comment]
@varffs i just tried the CMB2 Gallery plugin
https://github.com/mustardBees/cmb-field-gallery
and it works like a charm





 
what about using that?

File: single-director.php (hunk 4 above)
Comment (varffs) [inline single-director.php]: sorry somehow read that as get src!

**Comment by varffs** [comment]
Yeah sure lets use the gallery plug, but we going to parse the shortcode ourselves or use gallery.php?

**Comment by mixcas** [comment]
@varffs it returns an array of id's

**Comment by varffs** [comment]
oh wow not even just a shortcode! perfect. which leads me to wonder why we have not been using it before? was there a reason cus i have tried it out once?

**Comment by mixcas** [comment]
@varffs the code it's old and have some issues, i fixed some in 7e994be but this plugin could be easly refactored

we should do our own version or somthn

**Comment by varffs** [comment]
stills section needs to be set display none by default

**Pull request merged**